FastSimulation/SimplifiedGeometryPropagator/plugins/Bremsstrahlung.cc

```diff
@@ -0,0 +1,247 @@
+    return;
+    }
+
+    // probability to radiate a photon
+    double bremProba = radLengths * (4./3. * std::log(1./xmin)
+                      - 4./3. * (1.-xmin)
+                      + 1./2. * (1.-xmin*xmin));
+    
+  
+    // Number of photons to be radiated.
+    unsigned int nPhotons = poisson(bremProba, random);
+    if(nPhotons == 0) 
+    {
+    return;
+    }
+
+    // Needed to rotate photons to the lab frame
+    double theta = particle.momentum().Theta();
+    double phi = particle.momentum().Phi();
+    
+    // Calculate energy of these photons and add them to the event
+    for(unsigned int i=0; i<nPhotons; ++i) 
+    {
+        // Throw momentum of the photon
+        math::XYZTLorentzVector photonMom = brem(particle, xmin, random);
+
+        // Check that there is enough energy left.
+        if(particle.momentum().E() - particle.momentum().mass() < photonMom.E()) break;
+
+        // Rotate to the lab frame
+        photonMom = ROOT::Math::RotationZ(phi) * (ROOT::Math::RotationY(theta) * photonMom);
+
+        // Add a photon
+        secondaries.emplace_back(new fastsim::Particle(22, particle.position(), photonMom));        
+        
+        // Update the original e+/-
+        particle.momentum() -= photonMom;
+    }
+}   
+
+
+math::XYZTLorentzVector
+fastsim::Bremsstrahlung::brem(fastsim::Particle & particle, double xmin, const RandomEngineAndDistribution & random) const 
+{
+    double xp=0;
+    double weight = 0.;
+  
+    do{
+        xp = xmin * std::exp ( -std::log(xmin) * random.flatShoot() );
+        weight = 1. - xp + 3./4.*xp*xp;
+    }while(weight < random.flatShoot());
+  
+  
+    // Have photon energy. Now generate angles with respect to the z axis 
+    // defined by the incoming particle's momentum.
+
+    // Isotropic in phi
+    const double phi = random.flatShoot() * 2. * M_PI;
+    // theta from universal distribution
+    const double theta = gbteth(particle.momentum().E(), fastsim::Constants::eMass, xp, random)
+                            * fastsim::Constants::eMass / particle.momentum().E();
+  
+    // Make momentum components
+    double stheta = std::sin(theta);
+    double ctheta = std::cos(theta);
+    double sphi   = std::sin(phi);
+    double cphi   = std::cos(phi);
+  
+    return xp * particle.momentum().E() * math::XYZTLorentzVector(stheta*cphi, stheta*sphi, ctheta, 1.);  
+}
+
+double
+fastsim::Bremsstrahlung::gbteth(const double ener,
+                const double partm,
+                const double efrac,
+                const RandomEngineAndDistribution & random) const 
+{
+    const double alfa = 0.625;
+    
+    const double d = 0.13*(0.8+1.3/Z_)*(100.0+(1.0/ener))*(1.0+efrac);
+    const double w1 = 9.0/(9.0+d);
+    const double umax = ener*M_PI/partm;
+    double u;
+    
+    do 
+    {
+        double beta = (random.flatShoot()<=w1) ? alfa : 3.0*alfa;
+        u = -std::log(random.flatShoot()*random.flatShoot())/beta;
+    }while (u >= umax);
+
+    return u;
+}
+
+
```

Comment:
I would remove method below (likely inefficient) and use instead existing standard one:

random.poissonShoot(ymu);

Reply:
Done.